### PR TITLE
[WIP] Ability to specify Terraform binary, disable redirection of stderr, Terragrunt methods

### PR DESCRIPTION
**Pull request by kamsz**
Related to #122.

### CODE REVIEW

File: modules/shell/command.go [hunk 1]
```diff
@@ -83,6 +90,29 @@ func RunCommandAndGetOutputE(t *testing.T, command Command) (string, error) {
 	return output, nil
 }
 
+// This function captures stdout while still printing it to the stdout of this Go program
+func readStdout(t *testing.T, stdout io.ReadCloser) (string, error) {
```
Comment (brikis98) [inline modules/shell/command.go]: Any way to avoid copy/paste with the `readStdoutAndStderr` method?

**Comment by kamsz** [comment]
@brikis98 No, not yet. My MBP broke yesterday and I'm still setting up a fresh one. I'll get back to this as soon as I'm finished.

**Comment by kamsz** [comment]
@brikis98 I've merged these functions into one. Right now I'm running tests.

File: modules/shell/command.go [hunk 2]
```diff
@@ -84,32 +91,24 @@ func RunCommandAndGetOutputE(t *testing.T, command Command) (string, error) {
...
-	stderrScanner := bufio.NewScanner(stderr)
+	scanners := []bufio.Scanner{}
+	for i := range outputs {
+		scanners = append(scanners, *bufio.NewScanner(outputs[i]))
+	}
 
-	for {
-		if stdoutScanner.Scan() {
-			text := stdoutScanner.Text()
+	for i := range scanners {
```
Comment (brikis98) [inline modules/shell/command.go]: Doesn't this just loop over each scanner once and exit? The original `for` loop kept going infinitely until none of the scanners had any more data or it hit an error.

**Comment by robh007** [comment]
I was just going to ask the same question. Be really good to be able to use terragrunt, since I'm using terragrunt more and more these days.

**Comment by jakauppila** [comment]
@kamsz Curious about the status of this as well.

**Comment by jakauppila** [comment]
@brikis98 Is there anything else other than the additional two tests from your last comment left for this PR? I may look to pick it up and finish it.

**Comment by brikis98** [comment]
Hi @jakauppila 👋 

I think the remaining things are:

1. Merge latest from master, which has moved on quite a bit in the last several months. Deal with any merge conflicts as necessary).
1. Deal with the loop issue in my comment (if it's still relevant after merge!).
1. Add tests.

Thank you! 🍺

**Comment by yorinasub17** [comment]
This has been picked up in https://github.com/gruntwork-io/terratest/pull/304 by @snoozan! Going to close this in favor of that PR. Further comments and discussions should be moved there.

**Pull request closed without merging**